industry_benchmarks/utils/fix_networks.py

```diff
@@ -0,0 +1,605 @@
+    # complex one!
+    errmsg = ("The transformation data is not available, "
+              "please contact the OpenFE team")
+    raise ValueError(errmsg)
+
+
+def get_transformation(pur: gufe.ProtocolUnitResult):
+    """
+    Get a transformation out of a PUR
+
+    Returns
+    -------
+    transform : gufe.Transformation
+      The transformation for this edge
+    phase : str
+      Either "complex" or "solvent" depending on the phase type.
+    """
+    # We a assume a single result
+    ru_keys = [k for k in pur["protocol_result"]["data"].keys()]
+    if len(ru_keys) > 1:
+        errmsg = "Too many keys in the Protocol Unit Result file"
+        raise ValueError(errmsg)
+
+    units = pur["protocol_result"]["data"][ru_keys[0]]
+
+    if len(units) > 1:
+        errmsg = "Too many units found"
+        raise ValueError(errmsg)
+
+    if "stateA" not in units[0]["inputs"]:
+        return None, None
+
+    stateA = ChemicalSystem.from_dict(units[0]["inputs"]["stateA"])
+    stateB = ChemicalSystem.from_dict(units[0]["inputs"]["stateB"])
+    ligmap = LigandAtomMapping.from_dict(units[0]["inputs"]["ligandmapping"])
+
+    if any([isinstance(comp, gufe.ProteinComponent) for comp in stateA.components.values()]):
+        phase = "complex"
+    else:
+        phase = "solvent"
+
+    name = f"{phase}_{ligmap.componentA.name}_{ligmap.componentB.name}"
+
+    transform = Transformation(
+        stateA=stateA,
+        stateB=stateB,
+        mapping=ligmap,
+        name=name,
+        protocol=None,
+    )
+
+    return transform, phase
+
+
+def _check_and_deduplicate_transforms(
+    transforms_dict: dict[str, list[Transformation]]
+):
+    """
+    Traverse through a dictionary of transformations keyed
+    by the transformation name.
+
+    Check:
+      * There are a total of 3 transformations per entry
+      * The 3 transformations are the same
+
+    Returns
+    -------
+    alchemical_network : gufe.AlchemicalNetwork
+      A deduplicated alchemical network.
+    """
+    transform_list = []
+
+    for t_name, t_list in transforms_dict.items():
+        if len(t_list) != 3:
+            # TODO: we can turn this into a warning message if it's too painful
+            errmsg = (
+                f"Too few transformations found for {t_name} "
+                "this indicates a partially completed set of results. "
+                "Please ensure that your input network is finished and "
+                "any reproducible partial failures have been removed."
+            )
+            raise ValueError(errmsg)
+        if not all(a == t_list[0] for a in t_list):
+            errmsg = (
+                f"Transfromations for {t_name} do not match "
+                "this may indicate some kind of corruption of the alchemical "
+                "network. Please contact the OpenFE team."
+            )
+            raise ValueError(errmsg)
+
+        transform_list.append(t_list[0])
+
+    return AlchemicalNetwork(transform_list)
```

Comment:
The AlchemicalNetwork may not have "complete" cycles, i.e. could only have the solvent or complex leg. When generating the LigandNetwork from this, this information gets lost and broken networks are not recognized.

Reply:
This was fixed in commit https://github.com/OpenFreeEnergy/IndustryBenchmarks2024/pull/155/commits/8f7131f194d76d17831c2f264d587c443a334e61

Reply:
If instead the user should remove result .json files from edges where only one leg finished, these two lines can be removed, though at least a warning should be thrown.